Source/WebCore/loader/cache/CachedResourceRequest.cpp

```diff
@@ -166,10 +166,14 @@ static String acceptHeaderValueForImageResource()
     return builder.toString();
 }
 
-String CachedResourceRequest::acceptHeaderValueFromType(CachedResource::Type type)
+String CachedResourceRequest::acceptHeaderValueFromType(CachedResource::Type type, ResourceRequest& request)
```

Comment:
Should take in a `LocalFrame*` too

Reply:
* `acceptHeaderValueFromType(type, m_resourceRequest, frame)` is called from `CachedResourceRequest::setAcceptHeaderIfNone(CachedResource::Type type, LocalFrame* frame)`
* Then `request.setAcceptHeaderIfNone(type, frame)` is called from `CachedResourceLoader::prepareFetch(CachedResource::Type type, CachedResourceRequest& request, LocalFrame* frame)`
* Then `prepareFetch(type, request, frame)` is called from `ResourceErrorOr<CachedResourceHandle<CachedResource>> CachedResourceLoader::requestResource(CachedResource::Type type, CachedResourceRequest&& request, ForPreload forPreload, ImageLoading imageLoading)`

which gets me 

```
The following build commands failed:
	CompileC /Users/karlcow/code/OpenSource/WebKitBuild/WebCore.build/Debug/WebCore.build/Objects-normal/arm64e/UnifiedSource256.o /Users/karlcow/code/OpenSource/WebKitBuild/Debug/DerivedSources/WebCore/unified-sources/UnifiedSource256.cpp normal arm64e c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'WebCore' from project 'WebCore')
In file included from /Users/karlcow/code/OpenSource/WebKitBuild/Debug/DerivedSources/WebCore/unified-sources/UnifiedSource256.cpp:6:
(1 failure)
/Users/karlcow/code/OpenSource/Source/WebCore/loader/cache/CachedResourceLoader.cpp:1036:33: error: no viable conversion from 'Ref<LocalFrame>' (aka 'WTF::Ref<WebCore::LocalFrame>') to 'LocalFrame *'
    prepareFetch(type, request, frame);
In file included from /Users/karlcow/code/OpenSource/WebKitBuild/Debug/DerivedSources/WebCore/unified-sources/UnifiedSource256.cpp:1:
In file included from /Users/karlcow/code/OpenSource/Source/WebCore/WebCorePrefix.h:165:
In file included from /Users/karlcow/code/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/HashMap.h:25:
In file included from /Users/karlcow/code/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/HashTable.h:33:
In file included from /Users/karlcow/code/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/HashTraits.h:26:
In file included from /Users/karlcow/code/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/HashFunctions.h:26:
In file included from /Users/karlcow/code/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/RefPtr.h:28:
/Users/karlcow/code/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/Ref.h:136:5: note: candidate function
    operator T&() const { ASSERT(m_ptr); return *PtrTraits::unwrap(m_ptr); }
In file included from /Users/karlcow/code/OpenSource/WebKitBuild/Debug/DerivedSources/WebCore/unified-sources/UnifiedSource256.cpp:6:
/Users/karlcow/code/OpenSource/Source/WebCore/loader/cache/CachedResourceLoader.cpp:902:112: note: passing argument to parameter 'frame' here
void CachedResourceLoader::prepareFetch(CachedResource::Type type, CachedResourceRequest& request, LocalFrame* frame)
1 error generated.
make: *** [d] Error 65
```

Reply:
I already said in my earlier review but you don't need to change prepareFetch() to take in a frame. `CachedResourceLoader::prepareFetch()` can just pass `protectedFrame().get()` to request.setAcceptHeaderIfNone().

Reply:
`CachedResourceLoader::protectedFrame()` already exists.

Reply:
Ha understood now. Sorry about that! Thanks a lot.